manila/share/drivers/netapp/dataontap/cluster_mode/lib_base.py

```diff
@@ -3334,24 +3351,17 @@ def _safe_change_replica_source(self, dm_session, replica,
                                                 new_source_replica,
                                                 replica_list,
                                                 is_flexgroup=is_flexgroup)
-        except exception.StorageCommunicationException:
-            replica['status'] = constants.STATUS_ERROR
-            replica['replica_state'] = constants.STATUS_ERROR
-            if is_dr:
-                replica['export_locations'] = []
-            msg = ("Failed to change replica (%s) to a SnapMirror "
-                   "destination. Replica backend is unreachable.")
-
-            LOG.exception(msg, replica['id'])
-            return replica
-        except netapp_api.NaApiError:
+        # Catch everything! The original source may be down,
```

Comment:
I am wondering is such a disaster situation, there is better way to handle it than error it out. Can we forcefully cut the old snapmirror relationships and elect the new replica? It might be helpful when there are more than two replicas